src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,174 @@
+package frc.robot.subsystems;
+
+import java.util.function.BooleanSupplier;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.Follower;
+import com.ctre.phoenix6.controls.PositionDutyCycle;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.math.controller.ElevatorFeedforward;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.Constants;
+import SushiFrcLib.SmartDashboard.PIDTuning;
+import SushiFrcLib.SmartDashboard.TunableNumber;
+
+public class Elevator extends SubsystemBase {
+  private static Elevator instance;
+  
+  private final TalonFX leftMotor;
+  private final TalonFX rightMotor;
+  private final DigitalInput limitSwitch;
+  
+  private PIDTuning pid;
+  private final TunableNumber setpoint;
+  private final ElevatorFeedforward ffd; // Down direction feedforward
+  private final ElevatorFeedforward ffu; // Up direction feedforward
+  
+  // State tracking
+  private boolean up;
+  private boolean resetElevator;
+  private boolean openLoop;
+  private boolean inSecondStage;
```

Comment:
rename this to be more indicative of what it's used for. It's doesn't really have anything to do with the elevator being in the second stage, it's that we want to slow down because of a high CG